source/opt/debug_info_manager.cpp

```diff
@@ -194,9 +194,20 @@ uint32_t DebugInfoManager::CreateDebugInlinedAt(const Instruction* line,
     line_number = line->GetSingleWordOperand(kOpLineOperandLineIndex);
 
     // If we need the line number as an ID, generate that constant now.
+    // Do not use constant manager which may attempt to invoke building
+    // of DefUseManager which is invalidated for inlining. The extra
+    // constants that may be generated here is likely not significant
+    // and will likely be cleaned up in later passes.
     if (line_number_type == spv_operand_type_t::SPV_OPERAND_TYPE_ID) {
```

Comment:
To avoid creating the same constant multiple times, I would use the constant manager when it is available.
Can we do the following:
```
if (context()->AreAnalysesValid(IRContext::Analysis::kAnalysisConstants & IRContext::Analysis::kAnalysisDefUse)) {
        uint32_t line_id =
          context()->get_constant_mgr()->GetUIntConst(line_number);
} else {
  // create it directly
}
```

Reply:
Well, it certainly is correct, although I don't know if it will ever have any effect. That is because CreateDebugInlinedAt is currently only called from out of inlining which invalidates DefUse almost immediately. I am not sure this function will ever be called from anywhere but the inliner.

If you still want it I will make the change. Let me know.